resources/js/features/game-list/components/WantToPlayGamesDataTable/DataTableRowActions.tsx

```diff
@@ -0,0 +1,85 @@
+import { useQueryClient } from '@tanstack/react-query';
+import type { Row } from '@tanstack/react-table';
+import { RxCross2 } from 'react-icons/rx';
+
+import { BaseButton } from '@/common/components/+vendor/BaseButton';
+import { toastMessage } from '@/common/components/+vendor/BaseToaster';
+import {
+  BaseTooltip,
+  BaseTooltipContent,
+  BaseTooltipTrigger,
+} from '@/common/components/+vendor/BaseTooltip';
+import { useAddToBacklogMutation } from '@/common/hooks/useAddToBacklogMutation';
+import { useRemoveFromBacklogMutation } from '@/common/hooks/useRemoveFromBacklogMutation';
+
+/**
+ * If the table row needs to have more than one action, it should go into a menu.
+ * @see https://ui.shadcn.com/examples/tasks
+ */
+
+interface DataTableRowActionsProps<TData> {
+  row: Row<TData>;
+}
+
+export function DataTableRowActions<TData>({ row }: DataTableRowActionsProps<TData>) {
+  const queryClient = useQueryClient();
+
+  const removeFromBacklogMutation = useRemoveFromBacklogMutation();
+
+  const addToBacklogMutation = useAddToBacklogMutation();
+
+  const rowData = row.original as { game?: App.Platform.Data.Game };
+  const gameId = rowData?.game?.id ?? 0;
+  const gameTitle = rowData?.game?.title ?? '';
+
+  const handleRestoreToBacklogClick = () => {
+    toastMessage.promise(addToBacklogMutation.mutateAsync(gameId), {
+      loading: 'Removing...',
+      success: () => {
+        // Trigger a refetch of the current table page data and bust the entire cache.
+        queryClient.invalidateQueries({ queryKey: ['data'] });
+
+        return `Restored ${gameTitle}!`;
+      },
+      error: 'Something went wrong.',
+    });
+  };
```

Comment:
How can something be restored? The button immediately disappears on click.

Reply:
You should see an Undo button on the toast that appears:

https://github.com/user-attachments/assets/883a052d-5f0e-41f8-8d32-91b5d33f4d1e

Do you see something different?

Reply:
I wasn't looking for a button on the popup. I suppose it's better than nothing.

Is five seconds enough time for the user to register their mistake and get their cursor to that button before the toast goes away?

Reply:
In my opinion, it probably is. I can extend it by another second or two if you don't feel it's sufficient. I'm worried about a lot of long-running toasts stacking up if the user is removing a bunch of games.